### PR TITLE
Update `RequestFactory`

### PR DESCRIPTION
**Pull request by seladb**
- Addresses #524 
- Convert the `scope` used in the `RequestFactory` from `dict` to `HTTPScope` (typed-dict)
- Add the missing parameters to the `scope`
- Removed documentation for `create_test_request` which was removed from the codebase

# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [x] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [x] Have you updated the reference documentation?

### CODE REVIEW

File: starlite/testing.py [hunk 1]
```diff
@@ -368,8 +384,14 @@ def _create_scope(
...
             auth=auth,
             query_string=urlencode(query_params, doseq=True).encode() if query_params else b"",
-            path_params=[],
+            path_params=path_params,
             client=(self.server, self.port),
+            state=state,
+            asgi=ASGIVersion(spec_version="3.0", version="3.0"),
+            http_version=http_version or "1.1",
+            raw_path=path.encode("ascii"),
+            route_handler=None,  # type: ignore[typeddict-item]
```
Comment (seladb) [inline starlite/testing.py:393]: I'm not sure what is the right way to let users choose the `route_handler`. As per my understanding, there is no easy way to extract the handler from the `app` object, and letting the users choose just any handler is probably not useful. Any idea how to implement that?

File: starlite/testing.py [hunk 2]
```diff
@@ -368,8 +384,14 @@ def _create_scope(
             user=user,
             auth=auth,
             query_string=urlencode(query_params, doseq=True).encode() if query_params else b"",
-            path_params=[],
+            path_params=path_params,
             client=(self.server, self.port),
+            state=state,
+            asgi=ASGIVersion(spec_version="3.0", version="3.0"),
```
Comment (seladb) [inline starlite/testing.py:390]: I'm not sure this is the right way to build `ASGIVersion`. Please let me know if there's a better way

File: starlite/testing.py (hunk 1 above)
Comment (infohash) [reply starlite/testing.py:393]: Is route handler required to construct scope?

Comment (seladb) [reply starlite/testing.py:393]: It's a non-optional parameter in [`BaseScope`](https://github.com/starlite-api/starlite/blob/d6313981c9129da2f88efb09a9d14424733c781b/starlite/types/asgi_types.py#L78) but not required to create the `Request` object

Comment (infohash) [reply starlite/testing.py:393]: Can you use `app.route_handler_method_map().keys()`. This will give you all the registered route handler instances.

Comment (seladb) [reply starlite/testing.py:393]: But what if the app starts with `route_handlers=[]` or if it starts with more than one `route_handlers`? Which one should we choose?

Comment (Goldziher) [reply starlite/testing.py:393]: i wouldnt auto pick a value - i would suggest the following:

1. create a default placeholder, and use it if the user doesnt pass anything.
2. allow the user to pass a handler in manually. Consider this:

```python
@get()
def handler() -> None:
   ...

request = RequestFactory().get(path="/", route_handler=handler)
```

Comment (infohash) [reply starlite/testing.py:393]: Using raw_path and method, you can resolve its associated route handler by using `app.route_handler_method_map`.

```python
app.route_handler_method_map['/test'][method]
```

Comment (seladb) [reply starlite/testing.py:393]: Should the handler be of type `RouteHandlerType`, as the type in [`BaseScope`](https://github.com/starlite-api/starlite/blob/d6313981c9129da2f88efb09a9d14424733c781b/starlite/types/asgi_types.py#L78)?

Comment (Goldziher) [reply starlite/testing.py:393]: yup

**Review by Goldziher — APPROVED** [review]

File: tests/utils/test_sequence.py [hunk 3]
```diff
@@ -16,6 +16,6 @@ def y() -> None:
         pass
 
     unique_functions = unique([x, x, y, y])
-    assert unique_functions == [x, y]
+    assert unique_functions == [x, y] or [y, x]
```
Comment (seladb) [inline tests/utils/test_sequence.py:19]: Fixing a flaky test (not related to this PR)

**Pull request merged**